### PR TITLE
Use less _is_categorical_dtype

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Rehash of https://github.com/rapidsai/cudf/pull/14942

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/_lib/csv.pyx [hunk 1]
```diff
@@ -554,7 +554,7 @@ cdef data_type _get_cudf_data_type_from_dtype(object dtype) except *:
     # TODO: Remove this work-around Dictionary types
     # in libcudf are fully mapped to categorical columns:
     # https://github.com/rapidsai/cudf/issues/3960
```
Comment (bdice) [inline python/cudf/cudf/_lib/csv.pyx:556]: This linked issue has been resolved. Is there an action to be taken here (i.e. removal of the workaround)?

**Review by bdice — APPROVED** [review]

**Review by vyasr — APPROVED** [review]

File: python/cudf/cudf/_lib/csv.pyx (hunk 1 above)
Comment (mroeschke) [reply python/cudf/cudf/_lib/csv.pyx:556]: It appears although the issue is closed, the latest PR that addressed this function notes that it's still awaiting libcudf handling of category type: https://github.com/rapidsai/cudf/pull/12571#issue-1538618723

**Comment by mroeschke** [comment]
In bd7c1266efbd8ce2998d6606e180e45d946ce639 I added a carve-out for `cudf.dtype` to accept `"hex"` as a data type. It seems to only be recognized by `read_csv`. Is/was there a desire to expand this throughout cudf?

File: python/cudf/cudf/_lib/csv.pyx [hunk 2]
```diff
@@ -554,7 +554,7 @@ cdef data_type _get_cudf_data_type_from_dtype(object dtype) except *:
...
-    if cudf.api.types._is_categorical_dtype(dtype):
+    if isinstance(dtype, cudf.CategoricalDtype) or dtype == "category":
```
Comment (galipremsagar) [inline python/cudf/cudf/_lib/csv.pyx:557]:
```suggestion
    if isinstance(dtype, cudf.CategoricalDtype) or (isinstance(dtype, str) and dtype == "category"):
```

**Review by galipremsagar — CHANGES_REQUESTED** [review]

File: python/cudf/cudf/_lib/csv.pyx (hunk 2 above)
Comment (mroeschke) [reply python/cudf/cudf/_lib/csv.pyx]: Thanks. Finally got around to incorporating this change

**Review by galipremsagar — APPROVED** [review]

**Pull request merged**